$NetBSD: patch-programs_Makefile,v 1.14 2024/07/23 06:22:37 adam Exp $

Remove OS_VERSION as it is defined by pkgsrc, and causes "No such file or directory" message.
Unlock install target. Use pkgsrc install scripts.

--- programs/Makefile.orig	2024-07-21 17:29:49.000000000 +0000
+++ programs/Makefile
@@ -55,10 +55,6 @@ CPPFLAGS += -I$(LIBDIR) -DXXH_NAMESPACE=
 
 include ../Makefile.inc
 
-OS_VERSION ?= $(UNAME) -r
-ifeq ($(TARGET_OS)$(shell $(OS_VERSION)),SunOS5.10)
-LDFLAGS  += -lrt
-endif
 
 ALLFLAGS  = $(CFLAGS) $(CPPFLAGS) $(LDFLAGS)
 
@@ -201,7 +197,7 @@ BINDIR      ?= $(exec_prefix)/bin
 bindir      ?= $(BINDIR)
 DATAROOTDIR ?= $(prefix)/share
 datarootdir ?= $(DATAROOTDIR)
-MANDIR      ?= $(datarootdir)/man
+MANDIR      ?= $(PREFIX)/$(PKGMANDIR)
 mandir      ?= $(MANDIR)
 MAN1DIR     ?= $(mandir)/man1
 man1dir     ?= $(MAN1DIR)
@@ -209,13 +205,13 @@ man1dir     ?= $(MAN1DIR)
 .PHONY: install
 install: lz4
 	@echo Installing binaries in $(DESTDIR)$(bindir)
-	$(MAKE_DIR) $(DESTDIR)$(bindir)/ $(DESTDIR)$(man1dir)/
-	$(INSTALL_PROGRAM) lz4$(EXT) $(DESTDIR)$(bindir)/lz4$(EXT)
+	$(BSD_INSTALL_PROGRAM_DIR) $(DESTDIR)$(bindir)/ $(DESTDIR)$(man1dir)/
+	$(BSD_INSTALL_PROGRAM) lz4$(EXT) $(DESTDIR)$(bindir)/lz4$(EXT)
 	$(LN_SF) lz4$(EXT) $(DESTDIR)$(bindir)/lz4c$(EXT)
 	$(LN_SF) lz4$(EXT) $(DESTDIR)$(bindir)/lz4cat$(EXT)
 	$(LN_SF) lz4$(EXT) $(DESTDIR)$(bindir)/unlz4$(EXT)
 	@echo Installing man pages in $(DESTDIR)$(man1dir)
-	$(INSTALL_DATA) lz4.1 $(DESTDIR)$(man1dir)/lz4.1
+	$(BSD_INSTALL_MAN) lz4.1 $(DESTDIR)$(man1dir)/lz4.1
 	$(LN_SF) lz4.1 $(DESTDIR)$(man1dir)/lz4c.1
 	$(LN_SF) lz4.1 $(DESTDIR)$(man1dir)/lz4cat.1
 	$(LN_SF) lz4.1 $(DESTDIR)$(man1dir)/unlz4.1
